File: Leetcode/1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int t) {
        unordered_map<int,int> m;
        vector<int> ans;
        for(int i=0; i<nums.size(); i++) {
            int x = t-nums[i];
            if(m.find(x) != m.end()) {
                ans = {m[x],i};
                break;
            }
            m.insert({nums[i],i});
        }
        return ans;
    }
};

File: Leetcode/Rust/1. Two Sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut m:HashMap<i32, i32> = HashMap::new();
        
        for (i, &x) in nums.iter().enumerate() {
            match m.get(&(target - x)) {
                Some(&j) => return vec![j, i as i32],
                None => m.insert(x, i as i32),
            };
        }
        
        return vec![];
    }
}
